src/lib.rs

```diff
@@ -1,45 +1,97 @@
 
+// ---- START CLIPPY CONFIG
+
+#![cfg_attr(all(not(test), feature="clippy"), warn(result_unwrap_used))]
+#![cfg_attr(feature="clippy", warn(unseparated_literal_suffix))]
+#![cfg_attr(feature="clippy", warn(wrong_pub_self_convention))]
+
+// Enable clippy if our Cargo.toml file asked us to do so.
+#![cfg_attr(feature="clippy", feature(plugin))]
+#![cfg_attr(feature="clippy", plugin(clippy))]
+
+#![warn(missing_copy_implementations,
+        trivial_numeric_casts,
+        trivial_casts,
+        unused_extern_crates,
+        unused_import_braces,
+        unused_qualifications)]
+#![cfg_attr(feature="clippy", warn(cast_possible_truncation))]
+#![cfg_attr(feature="clippy", warn(cast_possible_wrap))]
+#![cfg_attr(feature="clippy", warn(cast_precision_loss))]
+#![cfg_attr(feature="clippy", warn(cast_sign_loss))]
+#![cfg_attr(feature="clippy", warn(missing_docs_in_private_items))]
+#![cfg_attr(feature="clippy", warn(mut_mut))]
+
+// Disallow `println!`. Use `debug!` for debug output
+// (which is provided by the `log` crate).
+#![cfg_attr(feature="clippy", warn(print_stdout))]
+
+// This allows us to use `unwrap` on `Option` values (because doing makes
+// working with Regex matches much nicer) and when compiling in test mode
+// (because using it in tests is idiomatic).
+#![cfg_attr(all(not(test), feature="clippy"), warn(result_unwrap_used))]
+#![cfg_attr(feature="clippy", warn(unseparated_literal_suffix))]
+#![cfg_attr(feature="clippy", warn(wrong_pub_self_convention))]
+
+// ---- END CLIPPY CONFIG
+
 #![cfg_attr(feature="nightly", feature(core_intrinsics))]
 
-extern crate text_layout;
 #[macro_use]
 extern crate limn_layout;
-extern crate cassowary;
-extern crate rusttype;
-extern crate glutin;
 #[macro_use]
 extern crate lazy_static;
-extern crate linked_hash_map;
-extern crate stable_bst;
 #[macro_use]
 extern crate maplit;
 #[macro_use]
 extern crate downcast_rs;
-extern crate euclid;
 #[macro_use]
 extern crate log;
+
+extern crate text_layout;
+extern crate cassowary;
+extern crate rusttype;
+extern crate glutin;
+extern crate linked_hash_map;
+extern crate stable_bst;
+extern crate euclid;
 extern crate webrender;
 extern crate gleam;
 extern crate app_units;
 extern crate image;
 
+/// Mouse & keyboard handling / delegating functions
 #[macro_use]
 pub mod event;
-pub mod app;
+/// Module for `WidgetBuilder` and callback handlers
 #[macro_use]
 pub mod widget;
+/// Module for layout / resizing handlers and layout solving
 #[macro_use]
 pub mod layout;
+
+/// Module for registering global handlers / window handling
+pub mod app;
+/// Module for various common widgets (button, text, canvas, etc.)
 pub mod widgets;
+/// Drawing functions for images, rectangles, text and ellipses
 pub mod draw;
+/// UI state handling, such as adding / removing widgets,
 pub mod ui;
+/// Points, rects and sizes definitions
 pub mod geometry;
+/// Font, image and texture resources
 pub mod resources;
+/// Color constants. TODO: use CSS instead
```

Comment:
I'm not sure why you assume this project will use CSS, styling will be overhauled and whatever is used may have some similarities with CSS, but I doubt it will actually be CSS, nor does that imply color constants will be removed. We can maybe discuss this in an issue later, I'll try and make a design proposal issue before implementing anything.

Reply:
I'm sorry, I assumed this prematurely. I originally meant to put a question mark behind this.

At some point in the future, I want to definitely style buttons (specifically padding and colors / gradients) to not be the default colors (I also think that the padding on the buttons is too much, but that's my taste). So, especially for the colors, there should be some function `from_hex()`, so that you can pass hex color codes into it.

Yes, I doubt that it will be CSS, but something alike. Like a "styling struct" that implements `Default` for the original values (like it is right now), but allows overriding the padding, etc. values. I have not explored `limn`s architecture enough to see if it would be easily possible, so I don't know how a design on this topic would look like. I've opened #22 for discussion of this topic.